cmd/generate-linux-fonts.rb

```diff
@@ -0,0 +1,174 @@
+    end
+
+    create_commits(:A, output_dir)
+    create_commits(:M, output_dir)
+    create_commits(:D, output_dir)
+  end
+
+  sig { returns(T::Hash[String, String]) }
+  def download_cask_data
+    file = Tempfile.new("cask.json")
+    download = URI.open("https://formulae.brew.sh/api/cask.json")
+    IO.copy_stream(download, file.path)
+    JSON.parse(file.read)
+  ensure
+    file.close
+    file.unlink
+  end
+
+  def create_commits(type, output_dir)
+    files = Utils.safe_popen_read("git", "diff", "--name-only", "--diff-filter=#{type}",
+                                  output_dir.to_s).strip.split("\n")
+    if type == :A
+      files += Utils.safe_popen_read("git", "ls-files", "--others", "--exclude-standard").strip.split("\n")
+    end
+
+    odebug files
+    files.each do |file|
+      token = file.gsub("#{output_dir}/", "").gsub(".rb", "")
+      message = case type
+      when :A
+        "#{token} (new font)"
+      when :M
+        "#{token} updated"
+      when :D
+        "#{token} deleted"
+      end
+
+      odebug "type (#{type}) for #{file}"
+      odebug Utils.safe_popen_read("git", "add", file)
+      odebug Utils.safe_popen_read("git", "commit", "-m", message)
+    end
+  end
+
+  sig { params(stanza: Symbol, value: T.untyped, args: T::Hash[String, String]).returns(String) }
+  def format_stanza(stanza, value, args = {})
+    return "" if value.nil?
+
+    if stanza == :sha256 && value == "no_check"
+      "\n  #{stanza} :#{value}"
+    elsif [:url, :head].include?(stanza)
+      "\n  #{stanza} #{format_url(value, args)}"
+    else
+      "\n  #{stanza} \"#{value.gsub("\"", "\\\"")}\""
+    end
+  end
+
+  def format_url(value, args)
+    ret = "\"#{value}\""
+    args.each do |key, arg_val|
+      ret += ",\n       #{key}: \"#{arg_val}\""
+    end
+
+    ret
+  end
+
+  def get_font_paths(cask)
+    paths = []
+
+    path_prefix = if cask["url_specs"].key?("only_path")
+      "#{cask["url_specs"]["only_path"]}/"
+    else
+      ""
+    end
+
+    cask["artifacts"].each do |artifact|
+      next unless artifact.key?("font")
+
+      artifact["font"].each do |font|
+        paths << "#{path_prefix}./**/#{font}"
+      end
+    end
+
+    paths
+  end
+
+  def generate_formula(cask)
+    classname = cask["token"].split("-").map(&:capitalize).join
+    formula = "class #{classname} < Formula"
+
+    formula += format_stanza :desc, cask["desc"] || "#{cask["name"].first&.capitalize} font"
+    formula += format_stanza :homepage, cask["homepage"]
+
+    if cask["version"] != "latest"
+      formula += format_stanza :url, cask["url"], cask["url_specs"]
+      formula += format_stanza :version, cask["version"]
+      formula += format_stanza :sha256, cask["sha256"]
+    end
+
+    formula += format_stanza :head, cask["url"], cask["url_specs"] if cask["version"] == "latest"
+
+    paths = get_font_paths(cask)
+    formula += "\n\n  depends_on :macos" if paths.count < 1
```

Comment:
May be nice to use `EOS` in these cases

Reply:
How do I make a heredoc with an inline `if`? My search skills are failing me.

Reply:
Two options:
- Append heredocs to each other
- Create variables containing heredocs or empty strings and concatenate them instead.

Just saves embedding `\n` and escaping `"`